pkg/scheduler/schedule_one.go

```diff
@@ -130,42 +130,41 @@ func (sched *Scheduler) schedulingCycle(
 	pod := podInfo.Pod
 	scheduleResult, err := sched.SchedulePod(ctx, fwk, state, pod)
 	if err != nil {
+
+		if err == ErrNoNodesAvailable {
+			status := framework.NewStatus(framework.UnschedulableAndUnresolvable).WithError(err)
+			return ScheduleResult{nominatingInfo: clearNominatedNode}, podInfo, status
+		}
+
 		// SchedulePod() may have failed because the pod would not fit on any host, so we try to
 		// preempt, with the expectation that the next time the pod is tried for scheduling it
 		// will fit due to the preemption. It is also possible that a different pod will schedule
 		// into the resources that were preempted, but this is harmless.
-		var (
-			nominatingInfo *framework.NominatingInfo
-			status         *framework.Status
-		)
 		if fitError, ok := err.(*framework.FitError); ok {
+			var nominatingInfo *framework.NominatingInfo
```

Comment:
to further reduce indentation, I suggest the following:

```
fitError, ok := err.(*framework.FitError)
if !ok {
  klog.ErrorS(err, "Error selecting node for pod", "pod", klog.KObj(pod))
  return ScheduleResult{nominatingInfo: clearNominatedNode}, podInfo, framework.AsStatus(err)
}

// the rest
```

Reply:
Yeah, that makes sense, updated.